File: Computabilidade-e-Complexidade-de-Algoritmos-main/java/16.java
public class AFDBlocosDeZeros {
    public static boolean blocosDeZeros(String s) {
        String estado = "q0";
        for (char c : s.toCharArray()) {
            if (estado.equals("q0") && c == '0') {
                estado = "q1";
            } else if (estado.equals("q1")) {
                if (c == '1') estado = "q2";
            } else if (estado.equals("q2") && c == '0') {
                return false;
            }
        }
        return true;
    }

    public static void main(String[] args) {
        System.out.println(blocosDeZeros("00111"));  // True
        System.out.println(blocosDeZeros("010"));    // False
    }
}

File: Computabilidade-e-Complexidade-de-Algoritmos-main/golang/16.go
package main
import "fmt"

func blocosDeZeros(s string) bool {
    estado := "q0"
    for _, char := range s {
        if estado == "q0" && char == '0' {
            estado = "q1"
        } else if estado == "q1" {
            if char == '1' {
                estado = "q2"
            }
        } else if estado == "q2" && char == '0' {
            return false
        }
    }
    return true
}

func main() {
    fmt.Println(blocosDeZeros("00111"))  // True
    fmt.Println(blocosDeZeros("010"))    // False
}
